File: python/discount_groups.py
import datetime
import sdk.python.entities
from sdk.python.entities import Property


class DiscountGroup(sdk.python.entities.Entity):

    resource = '/discount_group/'
    json_name = 'discountGroup'

    id = Property(int)
    created = Property(datetime.datetime)
    date_end = Property(datetime.datetime)
    date_start = Property(datetime.datetime)
    discount_type = Property(int)
    group_restricted = Property(bool)
    name = Property(str)
    discounts = Property("Discount")
    domain = Property("Domain")
    product = Property("Product")


class DiscountGroups(sdk.python.entities.Resource):

    entity_class = DiscountGroup
    json_name = 'discountGroups'


discountGroups = DiscountGroups()

File: php/src/discount_groups.php
<?php
declare(strict_types=1);

require_once 'entity.php';
require_once 'products.php';

class DiscountGroups extends Entity
{
    public static $resource = '/discount_groups/';
    public static $json_name = 'discount_group';

    public function __construct()
    {
        $this->json_property('id', 'integer');
        $this->json_property('archived', 'integer');
        $this->json_property('discount_type', 'integer');
        $this->json_property('name', 'string');
        $this->json_property('product', 'Product', $many = False, $recursive = True);
        $this->json_property('discounts', 'Discount', $many = True, $recursive = True);
    }
}


